core/src/main/java/com/novoda/noplayer/exoplayer/ExoPlayerTwoImpl.java

```diff
@@ -13,74 +22,95 @@
 import com.novoda.noplayer.VideoContainer;
 import com.novoda.noplayer.VideoDuration;
 import com.novoda.noplayer.VideoPosition;
+import com.novoda.noplayer.exoplayer.forwarder.ExoPlayerForwarder;
 import com.novoda.noplayer.player.PlayerInformation;
 
 import java.util.Collections;
 import java.util.List;
 
 public class ExoPlayerTwoImpl extends PlayerListenersHolder implements Player {
 
-    private final ExoPlayerTwoFacade facade;
+    private static final boolean RESET_POSITION = true;
+    private static final boolean RESET_STATE = false;
+
+    private final SimpleExoPlayer exoPlayer;
+    private final MediaSourceFactory mediaSourceFactory;
+    private final ExoPlayerForwarder forwarder;
     private final Heart heart;
-    private final OnPrepareForwarder prepareForwarder;
+
     private VideoContainer videoContainer = VideoContainer.empty();
+    private int videoWidth;
+    private int videoHeight;
 
     public static ExoPlayerTwoImpl newInstance(Context context) {
-        ExoPlayerTwoFacade facade = ExoPlayerTwoFacade.newInstance(context);
+        DefaultDataSourceFactory defaultDataSourceFactory = new DefaultDataSourceFactory(context, "user-agent");
+        Handler handler = new Handler(Looper.getMainLooper());
+        MediaSourceFactory mediaSourceFactory = new MediaSourceFactory(defaultDataSourceFactory, handler);
 
-        return new ExoPlayerTwoImpl(facade);
+        DefaultTrackSelector trackSelector = new DefaultTrackSelector();
+        SimpleExoPlayer exoPlayer = ExoPlayerFactory.newSimpleInstance(new DefaultRenderersFactory(context), trackSelector, new DefaultLoadControl());
+        return new ExoPlayerTwoImpl(exoPlayer, mediaSourceFactory, new ExoPlayerForwarder());
     }
 
-    private ExoPlayerTwoImpl(ExoPlayerTwoFacade facade) {
-        this.facade = facade;
+    private ExoPlayerTwoImpl(SimpleExoPlayer exoPlayer, MediaSourceFactory mediaSourceFactory, ExoPlayerForwarder exoPlayerForwarder) {
+        this.exoPlayer = exoPlayer;
+        this.mediaSourceFactory = mediaSourceFactory;
+        forwarder = exoPlayerForwarder;
         Heart.Heartbeat<Player> onHeartbeat = new Heart.Heartbeat<>(getHeartbeatCallbacks(), this);
```

Comment:
I think we have done this in a follow-up branch that we are waiting to open 😆  but yes we should try and move all of these out of the constructor.